internal/ent/schema/vm.go

```diff
@@ -0,0 +1,67 @@
+package schema
+
+import (
+	"entgo.io/contrib/entgql"
+	"entgo.io/ent"
+	"entgo.io/ent/schema"
+	"entgo.io/ent/schema/field"
+	"entgo.io/ent/schema/index"
+
+	"go.infratographer.com/x/entx"
+	"go.infratographer.com/x/gidx"
+)
+
+type VirtM struct {
+	ent.Schema
+}
+
+func (VirtM) Mixin() []ent.Mixin {
+	return []ent.Mixin{
+		entx.NewTimestampMixin(),
+	}
+}
+
+func (VirtM) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("id").
+			GoType(gidx.PrefixedID("")).
+			Unique().
+			Immutable().
+			Comment("The ID of the VirtM.").
+			Annotations(
+				entgql.OrderField("ID"),
+			).
+			DefaultFunc(func() gidx.PrefixedID { return gidx.MustNewID(VirtMPrefix) }),
+		field.Text("hostname").
+			NotEmpty().
+			Comment("The name of the VirtM.").
+			Annotations(
+				entgql.QueryField(),
+				entgql.Type("String"),
+				entgql.OrderField("HOSTNAME"),
+			),
```

Comment:
I'm not sure this should be `hostname` but rather just `name`. We would likely set `hostname` from cloud-init, and I'm (at this moment) not sure we would be enriching a cloud-init metadata file instead of expecting to be provided with all relevant information. In essence I think we would prefer clients handle the creation of cloud-init specs.